azuredevops/internal/service/git/resource_git_repository.go

```diff
@@ -282,18 +290,68 @@ func resourceGitRepositoryUpdate(d *schema.ResourceData, m interface{}) error {
 		return fmt.Errorf("Error converting terraform data model to AzDO project reference: %+v", err)
 	}
 
+	// you cannot update a disabled repo. We must temporarily enable to update
+	repoActual, err := gitRepositoryRead(clients, repo.Id.String(), *repo.Name, projectID.String())
+	if err != nil {
+		if utils.ResponseWasNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error looking up repository with ID %s and Name %s. Error: %v", *repo.Id, *repo.Name, err)
+	}
+
+	if *repoActual.IsDisabled {
+		err = updateIsDisabledGitRepository(clients, repo.Id.String(), projectID.String(), false)
+		if err != nil {
+			return fmt.Errorf(" enabling created repository : %+v", err)
+		}
+	}
```

Comment:
It's better to raise an error than to enable the repo and update it.

Reply:
@xuzhang3 I have updated the code per your recommendation to error instead of simply enabling the repo. The code is ready for another review.